openlibrary/plugins/openlibrary/js/initAnnouncementBanner.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Init the announcement banners if any
+ * If patron has dismissed the banner by clicking the 'X' button, banner storage key will be added to localstorage
+ * Banners with storage keys within the localstorage will not show again
+ */
+export function initAnnouncementBanner() {
+    const announcementBanner = document.getElementById('announcement-banner');
+    const storedHiddenBannerKeys = localStorage.getItem('hiddenBannerKeys');
+    const bannerKey = announcementBanner.getAttribute('data-storage-key');
+    if (storedHiddenBannerKeys) {
+        const hiddenBannerKeys = JSON.parse(storedHiddenBannerKeys);
+        if (hiddenBannerKeys[bannerKey] === true) {
+            announcementBanner.style.display = 'none';
```

Comment:
Add class `hidden` to the announcement banner to hide it.

Reply:
@jimchamp Seems like adding a class (`announcementBanner.classList.add('hidden')`) is slower than setting the visibility style directly (`announcementBanner.style.display = 'none';`). In the former, the banner visibly renders for a split second before being hidden. Would that be ideal?